src/edge/simulator.ts

```diff
@@ -226,10 +196,156 @@ export class Simulator {
         });
     }
 
+    private async getSimulatorType(): Promise<SimulatorType> {
+        if (await this.simulatorInstalled()) {
+            if (os.platform() === "win32") {
+                return SimulatorType.Standalone;
+            } else {
+                return SimulatorType.Pip;
+            }
+        }
+        return SimulatorType.NotInstalled;
+    }
+
+    private async getLatestSimulatorVersion(): Promise<string | undefined> {
+        try {
+            let version: string;
+            if (os.platform() === "win32") {
+                const userConfiguredVersion: string = this.getUserConfiguredVersion();
+                if (userConfiguredVersion) {
+                    version = userConfiguredVersion;
+                } else {
+                    const releaseInfoJson: any = await Simulator.getLastestStandaloneSimulatorInfo();
+                    version = releaseInfoJson.tag_name;
+                }
+            } else {
+                const pipResponse: string = await request.get(Simulator.iotedgehubdevVersionUrl);
+                version = JSON.parse(pipResponse).info.version;
+            }
+            return version;
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    private getUserConfiguredVersion(): string {
+        const version: string = Configuration.getConfigurationProperty("simulator.version");
+        if (version && version.match(/^\d+\.\d+\.\d+$/)) {
+            return "v" + version;
+        }
+    }
+
+    private async getCurrentSimulatorVersion(): Promise<string | undefined> {
+        const output: string = await Executor.executeCMD(undefined, this.getSimulatorExecutorPath(true), { shell: true }, "--version");
+        const version: string | null = Simulator.extractVersion(output);
+        return version;
+    }
+
+    private async simulatorInstalled(): Promise<boolean> {
+        return await Simulator.checkCmdExist(this.getSimulatorExecutorPath(true));
+    }
+
+    private getSimulatorExecutorPath(forceUseCmd: boolean = false): string {
```

Comment:
the return value of this method better to be cached in the class instance. We call this method everyplace to call iotedgehub related stuff. Besides, I think check the version on Activate is more cleaner.